R/doc-params.R

```diff
@@ -1,31 +1,117 @@
+#' Argument type: data-masking
+#'
+#' @description
+#' This page describes the `<data-masking>` argument modifier which
+#' indicates that the argument uses **data masking**, a sub-type of
+#' tidy evaluation. If you've never heard of tidy evaluation before,
+#' start by reading `vignette("programming")` and
+#' <https://r4ds.hadley.nz/functions.html#data-frame-functions>.
+#'
+#' # Key techniques
+#'
+#' *  If you want the user to supply the variable (or function of variables)
+#'    in a function argument, embrace the argument, e.g. `filter(df, {{ var }})`.
+#'
+#'    ```R
+#'    dist_summary <- function(df, var) {
+#'      df %>%
+#'        summarise(n = n(), min = min({{ var }}), max = max({{ var }}))
+#'    }
+#'    mtcars %>% dist_summary(mpg)
+#'    mtcars %>% group_by(cyl) %>% dist_summary(mpg)
+#'    ```
+#'
+#' *  If you have the column name as a character vector, use the `.data`
+#'    pronoun, e.g. `summarise(df, mean = mean(.data[[var]]))`.
+#'
+#'    ```R
+#'    for (var in names(mtcars)) {
+#'      mtcars %>% count(.data[[var]]) %>% print()
+#'    }
+#'
+#'    lapply(names(mtcars), function(var) mtcars %>% count(.data[[var]]))
+#'    ```
+#'
+#'    (Note that the contents of `[[`, e.g. `var` above, is never evaluated
+#'    in the data environment so you don't need to worry about a data-variable
+#'    called `var` causing problems.)
```

Comment:
This paragraph is a bit unclear. Maybe just say something along the lines of:

> One advantage of subsetting with `.data[[var]]` is that it doesn't suffer from the problem of data-mask ambiguity. See https://rlang.r-lib.org/reference/topic-data-mask-ambiguity.html for more information about this.

Reply:
Maybe we can just not mention it at all here.